packages/data-provider/src/services/submissionService.ts

```diff
@@ -305,29 +312,34 @@ const service = (dependencies: BaseDependencies) => {
 			}
 
 			const entitiesToProcess: string[] = [];
+			const submissionsToValidate: (NewSubmittedData | SubmittedData)[] = [];
 
 			const submittedDataArray = await getSubmittedDataByCategoryIdAndOrganization(
 				categoryId,
 				submission?.organization,
 			);
 
-			const submissionsToValidate = Object.entries(submission.data).flatMap(([entityName, submissionData]) => {
-				entitiesToProcess.push(entityName);
-				return submissionData.records.map((record) => {
-					const newSubmittedData: NewSubmittedData = {
-						data: record,
-						dictionaryCategoryId: categoryId,
-						entityName,
-						isValid: false, // By default New Submitted Data is created as invalid until validation process proves otherwise
-						organization: submission.organization,
-						originalSchemaId: submission.dictionaryId,
-						lastValidSchemaId: submission.dictionaryId,
-						systemId: generateIdentifier(entityName, record),
-						createdBy: '', // TODO: get User from auth
-					};
-					return newSubmittedData;
+			if (submission.data?.inserts) {
```

Comment:
So far we only store new data on submission. Future PR's we're going to add `updates` and `deletes` to be included to perform validation